torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -122,15 +122,6 @@ void TensorPipeAgent::pipeRead(
 
     // Allocate memory and fill in pointers
     Message rpcMessage = tensorpipeAllocateMessage(tpMessage);
-    TORCH_INTERNAL_ASSERT(
-        rpcMessage.tensors().size() == tpMessage.tensors.size(),
```

Comment:
Looks like `tensorpipeAllocateMessage` does add tensors into `rpcMessage`, why removing this check here?

Reply:
Same reason as above, the number of tensors in the TP message is an implementation detail and may not be equal to the number of tensors in the RPC message.

Reply:
Right now tensorpipe id not deduplicating tensors sharing the same storage. I think the number of tensors between RPC message and TensorPipe message are always the same. Let's keep these checks until handling duplicated tensor storage?

Reply:
Yes, it is true that at the moment that condition would be true and the check would pass. My take is that tests should only be allowed to check what is in the specification of a function, i.e., the semantics of its interface. If they do instead check the implementation then they may need to be updated each time the code changes which means they are not robust and don't really provide any safeguard.

Reply:
This guard is right to make sure each update of the implementation doesn't break the critical assumption, unless we start to handle the duplicated tensor storage, which will break the assumption.

Reply:
The only "critical" assumption, IMO, is that deserializing a serialized message gives back the original. Which is now exactly the only thing the tests are checking.